File: Oidc.AuthorizationServer/ViewModels/ResetPasswordViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.ViewModels
{
    public class ResetPasswordViewModel
    {
        [Required]
        public string OldPassword { get; set; }

        [Required]
        public string NewPassword { get; set; }
    }
}

File: oidc/src/services/reset-password/reset-password-view-model.ts
import { Required } from '@lvl/front-end';

/**
 *  What the user wants their password to be, and what it was (for validation).
 */
export class ResetPasswordViewModel {
    /** The current password, to be compared before updating. */
    @Required() oldPassword: string;

    /** The password that the user wants. */
    @Required() newPassword: string;
}
